hera_mc/cm_active.py

```diff
@@ -0,0 +1,227 @@
+            session = db.sessionmaker()
+        self.session = session
+        self.at_date = cm_utils.get_astropytime(at_date)
+        self.parts = None
+        self.connections = None
+        self.info = None
+        self.apriori = None
+
+    def is_data_current(self, data_type, at_date):
+        """
+        Determines if the class data of type data_type is current at at_date.
+
+        Parameters
+        ----------
+        data_type : str
+            One of the allowed data_types:  parts, connections, info, apriori
+        at_date : astropytime.Time or None
+            Date for which to check.  If none, assumes match to self.at_date
+
+        Returns
+        -------
+        bool
+            True if current
+        """
+
+        if getattr(self, data_type) is None:
+            return False
+        if at_date is None or abs(at_date.gps - self.at_date.gps) < self.close_enough:
+            return True
+        return False
+
+    def set_times(self, at_date):
+        """
+        Makes sure that at_date and self.at_date are synced and supplies gps time
+
+        Parameters
+        ----------
+        at_date : astropytime.Time or None
+            Date for which to check.  If none, assumes self.at_date
+
+
+        Returns
+        -------
+        int
+            gps seconds of at_date
+        """
+        if at_date is not None:
+            self.at_date = at_date
+        return self.at_date.gps
+
+    def load_parts(self, at_date=None):
+        """
+        Retrieves all active parts for a given at_date.
+        Sets this object's at_date and loads the active parts on to the object.
+        If at_date is None, the existing at_date on the object will be used.
+
+        Writes class dictionary:
+                self.parts - keyed on part:rev
+
+        Parameters
+        ----------
+        at_date : str, int, float, Time, datetime (optional)
+            The date for which to check as active, given as anything comprehensible
+            to get_astropytime.  If not present uses self.at_date
+        """
+        at_date = cm_utils.get_astropytime(at_date)
+        if self.is_data_current('parts', at_date):
+            return
+        gps_time = self.set_times(at_date)
+        self.parts = {}
+        for prt in self.session.query(partconn.Parts).filter((partconn.Parts.start_gpstime <= gps_time)
+                                                             & ((partconn.Parts.stop_gpstime >= gps_time)
+                                                             | (partconn.Parts.stop_gpstime == None))):  # noqa
+            key = cm_utils.make_part_key(prt.hpn, prt.hpn_rev)
+            self.parts[key] = prt
+
+    def load_connections(self, at_date=None):
+        """
+        Retrieves all active connections for a given at_date.  If a part:rev:port
+        connection already exists, it will generate an error.
+        If at_date is None, uses self.at_date.  If not, will redefine self.at_date
```

Comment:
This has the same weird wording that `load_parts` had earlier. Can you make the same change here and on `get_info` (which should be `load_info`?) and `load_apriori`?